From 8f3958fe33f541c01ec7b19d33cf6f927e311ca7 Mon Sep 17 00:00:00 2001
From: David Drysdale <drysdale@google.com>
Date: Tue, 17 May 2016 17:50:54 +0100
Subject: [PATCH] Fix year checking for GeneralizedTime

RFC5280 s4.1.2.5: "certificate validity dates in 2050 or later MUST
be encoded as GeneralizedTime".
---
 asn1_time.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/asn1_time.c b/asn1_time.c
index 0507fa7b4a03..158d87c36a38 100644
--- a/asn1_time.c
+++ b/asn1_time.c
@@ -44,10 +44,11 @@ static bool time_to_tm(ASN1_TIME *t, bool general, struct tm *tm)
 	int i = 0;
 	if (general)
 	{
-		tm->tm_year = (s[0] - '0') * 1000 + (s[1] - '0') * 100 + (s[2] - '0') * 10 + s[3] - '0' - 1900;
+		int year = ((s[0] - '0') * 1000 + (s[1] - '0') * 100 + (s[2] - '0') * 10 + s[3] - '0');
+		tm->tm_year = year - 1900;
 		i += 4;
 
-		if (tm->tm_year < 150 || tm->tm_year >= 50)
+		if (year < 2050)
 		{
 			return false;
 		}
-- 
2.8.0.rc3.226.g39d4020

